doc/sphinxext/custom_autosummary_new_suffix.py

```diff
@@ -70,12 +69,24 @@ def setup(app):
     app.add_config_value(
         'custom_autosummary_generated_dirname', '', None)
 
-    # Override process_generate_options added by sphinx.ext.autosummary
+    # Find listener id for process_generate_options added by
+    # sphinx.ext.autosummary
+    process_generate_options_id = None
+
     builder_inited_listeners = app.events.listeners["builder-inited"]
+    # sphinx < 3.0, builder_inited_listeners is a dict
+    if isinstance(builder_inited_listeners, dict):
+        for listener_id, obj in builder_inited_listeners.items():
+            if (inspect.isfunction(obj)
+                    and obj.__name__ == "process_generate_options"):
+                process_generate_options_id = listener_id
+                break
```

Comment:
do we need to support older sphinx?

Reply:
I think as long as we put a nicer error message stating "Please update to sphinx > 3.0", would be good enough for developers.